src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let currentResult = 0;
+
+  return {
+    get result() {
+      return currentResult;
+    },
+    add(value) {
+      currentResult += value;
+    },
+    subtract(value) {
+      currentResult -= value;
+    },
+    multiply(value) {
+      currentResult *= value;
+    },
+    divide(value) {
+      if (value !== 0) {
+        currentResult /= value;
+      }
+    },
+    reset() {
+      currentResult = 0;
+
+      return this;
+    },
+    operate(callback, value) {
+      callback.call(this, value);
+
+      return this;
+    },
+  };
```

Comment:
[CODE KNOWLEDGE] - When defining methods inside an object, you don't need to use the function keyword. Use the method shorthand syntax instead. For example, instead of `add: function(value) { ... },` you can simply write `add(value) { ... },`.